### PR TITLE
#1352 - Disambiguation for lazy intersection of polyhedra 

### LINKED ISSUE #1352: MethodError in overapproximation of lazy intersections of polyhedra
```julia
julia> P = rand(HPolyhedron);

julia> overapproximate(P ∩ P, OctDirections(2));

MethodError: LazySets.Approximations.overapproximate(::Intersection{Float64,HPolyhedron{Float64},HPolyhedron{Float64}}, ::OctDirections{Float64}) is ambiguous. Candidates:
  overapproximate(cap::Intersection{N,#s35,#s24} where #s24<:LazySet where #s35<:AbstractPolyhedron{N}, dir::LazySets.Approximations.AbstractDirections{N}) where N<:Real in LazySets.Approximations at /Users/forets/.julia/dev/LazySets/src/Approximations/overapproximate.jl:442
  overapproximate(cap::Intersection{N,#s35,#s24} where #s24<:AbstractPolyhedron{N} where #s35<:LazySet, dir::LazySets.Approximations.AbstractDirections{N}) where N<:Real in LazySets.Approximations at /Users/forets/.julia/dev/LazySets/src/Approximations/overapproximate.jl:432
Possible fix, define
  overapproximate(::Intersection{N<:Real,#s35,#s24} where #s24<:AbstractPolyhedron{N<:Real} where #s35<:AbstractPolyhedron{N<:Real}, ::LazySets.Approximations.AbstractDirections{N<:Real})

Stacktrace:
 [1] top-level scope at In[66]:2
```

### PR DESCRIPTION
**Pull request by mforets**
Closes #1352.

Marking this as WIP since i consider it still under discussion.

### CODE REVIEW

File: src/Approximations/overapproximate.jl [hunk 1]
```diff
@@ -442,6 +442,17 @@ function overapproximate(cap::Intersection{N,
     return overapproximate_cap_helper(cap.Y, cap.X, dir; kwargs...)
 end
 
+# disambiguation
+function overapproximate(cap::Intersection{N,
+                                           <:AbstractPolyhedron{N},
+                                           <:AbstractPolyhedron},
```
Comment (schillic) [inline src/Approximations/overapproximate.jl]:
```suggestion
                                           <:AbstractPolyhedron{N}},
```

**Comment by schillic** [comment]
Maybe we should assert boundedness of `X` in `overapproximate_cap_helper` (compare to #1517) or change the return type of `Q` accordingly.

https://github.com/JuliaReach/LazySets.jl/blob/b451d0a8fdfaa6df12ad7ff60f744481db331886/src/Approximations/overapproximate.jl#L360-L380

File: src/Intersection.jl [hunk 2]
```diff
@@ -439,6 +438,34 @@ function ρ(d::AbstractVector{N}, cap::Intersection{N, S1, S2}; kwargs...
...
+### Output
+
+The support function for the given direction.
+
+### Algorithm
+
+We combine the constraints of the two polyhedra to a new `HPolyhedron`, for
+which we then evaluate the support function.
+"""
+function ρ(d::AbstractVector{N}, cap::Intersection{N, S1, S2}; kwargs...
```
Comment (mforets) [inline src/Intersection.jl]: i dont remember if `kwargs` here is intentional or can be removed

Comment (schillic) [reply src/Intersection.jl]: Other methods have that, so we should keep it for consistency of the interface (although that is already quite messed up).

**Pull request merged**